pkg/collector/corechecks/servicediscovery/events.go

```diff
@@ -0,0 +1,132 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2016-present Datadog, Inc.
+
+package servicediscovery
+
+import (
+	"context"
+	"encoding/json"
+
+	"github.com/DataDog/datadog-agent/comp/core/hostname"
+	"github.com/DataDog/datadog-agent/comp/core/hostname/hostnameimpl"
+	"github.com/DataDog/datadog-agent/comp/forwarder/eventplatform"
+	"github.com/DataDog/datadog-agent/pkg/aggregator/sender"
+	pkgconfig "github.com/DataDog/datadog-agent/pkg/config"
+	"github.com/DataDog/datadog-agent/pkg/util/log"
+)
+
+type eventType string
+
+const (
+	eventTypeStartService     = "start-service"
+	eventTypeEndService       = "end-service"
+	eventTypeHeartbeatService = "heartbeat-service"
+)
+
+type eventPayload struct {
+	ApiVersion          string    `json:"api_version"`
+	NamingSchemaVersion string    `json:"naming_schema_version"`
+	RequestType         eventType `json:"request_type"`
+	ServiceName         string    `json:"service_name"`
+	HostName            string    `json:"host_name"`
+	Env                 string    `json:"env"`
+	ServiceLanguage     string    `json:"service_language"`
+	ServiceType         string    `json:"service_type"`
+	StartTime           int64     `json:"start_time"`
+	LastSeen            int64     `json:"last_seen"`
+	APMInstrumentation  bool      `json:"apm_instrumentation"`
```

Comment:
This field should be of type string, with three legal values:

- none
- packaged
- injected

I need to extract code from the injection library to provide these values.